packages-content-model/roosterjs-content-model-core/lib/corePlugin/ContentModelCopyPastePlugin.ts

```diff
@@ -250,6 +256,41 @@ class ContentModelCopyPastePlugin implements PluginWithState<CopyPastePluginStat
     };
 }
 
+/**
+ * @internal
+ * Exported only for unit testing
+ */
+export function adjustSelectionForCopyCut(pasteModel: ContentModelDocument) {
+    let selectionMarker: ContentModelSelectionMarker | undefined;
+    let firstBlock: ContentModelBlock | undefined;
+
+    iterateSelections(pasteModel, (_, tableContext, block, segments) => {
+        if (selectionMarker) {
+            if (
+                tableContext &&
```

Comment:
Not sure if it is a valid scenario but I feel we should also need to save the first tableContext, then just need to check this tableContext is different than the first one. And tableContext can be null here. That will cover the case like:

```html
<table><tr><td>
text [SelectionMarker]
</td></tr></table>
[Selected: text.... ]
```

Reply:
Consider add test case for this case

Reply:
Ok, I think we can do that to.